lib/housing-available/housing-available.controllers.js

```diff
@@ -1 +1,229 @@
+              severity: 'Notice',
+              request: request,
+            });
+          } catch (e) { }
+          throw Boom.badData('No results for that address');
+        }
+        const firstGeocode = geocodes.results[0];
+
+        // const util = require('util')
+        // console.log(util.inspect(firstGeocode, {showHidden: false, depth: null}))
+        // Extract address info from associate address object, if one exists;
+        addressObject = postalAddressLib.extractAddress(firstGeocode);
+        if (!addressObject.addressNumber) {
+          try {
+            await Log.request({
+              type: 'property.submit.failed.street_number',
+              severity: 'Notice',
+              request: request,
+            });
+          } catch (e) { }
+          throw Boom.badData('Address is missing a street number');
+        }
+        // In some places addresses don't have a county; in that case, copy up the locality to normalize
+        if (!addressObject.hasOwnProperty('addressCounty')) {
+          addressObject.addressCounty = addressObject.addressLocality;
+        }
+        if (
+          addressObject.addressRegion !== process.env.ADDRESS_LIMIT_STATE ||
+          addressObject.addressCounty !== process.env.ADDRESS_LIMIT_COUNTY
+        ) {
+          try {
+            await Log.request({
+              type: 'property.submit.failed.county',
+              severity: 'Notice',
+              request: request,
+            });
+          } catch (e) { }
+          throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`);
+        }
+        const addressHash = postalAddressLib.hashAddress(addressObject);
+
+        addressObject.hash = addressHash;
+
+        const existingAddresses = await postalAddressLib.getAddressesByHash(addressHash);
+        if (existingAddresses.length > 0) {
+          housingAvailableObject.PropertyId = existingAddresses[0].PropertyId;
+        } else {
+          // TODO: create new Property and attach it to this;
+        }
+        // let locationTerms = [addressObject.addressRegion];
+        let locationTerms = [addressObject.addressRegion, addressObject.addressLocality];
+        if (addressObject.hasOwnProperty('addressNeighborhood')) {
+          locationTerms.push(addressObject.addressNeighborhood);
+        } else {
+          locationTerms.push(addressObject.addressLocality);
+        }
+        let locationId = await locationLib.locationFindCreateNestedTerms(locationTerms);
+        housingAvailableObject.LocationId = locationId;
+      }
+
+      if (hasAddress) {
+        housingAvailableObject.address = request.payload.address;
+      }
+
+      if (request.payload.contact) {
+        housingAvailableObject.contact = request.payload.contact;
+      }
+
+      if (request.payload.metadata) {
+        housingAvailableObject.metadata = request.payload.metadata;
+      }
```

Comment:
metadata isn't on the payload. and users can't/shouldn't be able to submit it (it's usually something calculated by the system to help the system display info without having to calculate things on the fly). please remove.